shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Genesis.hs

```diff
@@ -111,8 +114,9 @@ data ShelleyGenesis c = ShelleyGenesis
     sgInitialFunds :: !(Map (Addr c) Coin),
     sgStaking :: !(ShelleyGenesisStaking c)
   }
-  deriving stock (Eq, Show, Generic)
-  deriving anyclass (NoUnexpectedThunks)
+  deriving stock (Eq, Show)
+
+deriving via UseIsNormalFormNamed "ShelleyGenesis" (ShelleyGenesis era) instance (Era era) => NoUnexpectedThunks (ShelleyGenesis era)
```

Comment:
Why `UseIsNormalFormNamed`? The downside of this compared to a generic instance is that when we find a thunk, its location will be `ShelleyGenesis`, instead of the precise field/type containing the thunk, e.g., `GenDelegPair`.

Reply:
I am going to put back all the contentious derivations you commented on to the way they were in master for now, and then will change them back as needed for TxOut parametrization

Reply:
Great! Thanks (There might be some more that I didn't comment on, but my opinion on them is the same :slightly_smiling_face:).